adapters/repos/db/vector/hnsw/index.go

```diff
@@ -1013,9 +1013,34 @@ func (h *hnsw) Stats() (common.IndexStats, error) {
 	return &stats, nil
 }
 
-func (h *hnsw) VectorStorageSize() int64 {
-	// TODO-usage: Implement this
-	return 0
+func (h *hnsw) VectorStorageSize(ctx context.Context) int64 {
+	h.RLock()
+	defer h.RUnlock()
+
+	// Always use the dimensions bucket for accurate counts instead of cache-based counts
+	// This ensures we get the correct total vectors and dimensions regardless of cache size
+	dimensions, objectCount := h.store.CalcTargetVectorDimensionsFromStore(ctx, "", func(dimLen int, v []lsmkv.MapPair) (int, int) {
```

Comment:
`Store::CalcTargetVectorDimensionsFromStore` returns `sum` (whatever that is) and `dimensions`.
is assignment `dimensions, objectCount := ...` correct?

Reply:
good catch i think it's swapped here